app/controllers/registration_controller.rb

```diff
@@ -22,13 +22,23 @@ def new
   end
 
   def destroy
-    @registration = Registration.find(params[:registration_id])
-    if current_user == @registration.user && @registration.event.start_date.future?
+    is_registrationId_given = params[:registration_id].present?
+    if is_registrationId_given
```

Comment:
You can just use user Id and Event_id form the params and find it. We don't really need this if else